feature-libs/cart/quick-order/core/services/quick-order.service.ts

```diff
@@ -114,6 +115,23 @@ export class QuickOrderService {
         }
       });
 
+    subscription.add(
+      this.eventService
+        .get(CartAddEntryFailEvent)
+        .subscribe((cartEvent: CartAddEntryFailEvent) => {
+          let event = { ...cartEvent } as CartAddEntrySuccessEvent;
```

Comment:
Why not just change return type of this function to `Observable<[number, Array<CartAddEntrySuccessEvent | CartAddEntryFailEvent>]>`?

Reply:
Providing a second type wouldn't help us. We would still be missing important fields and need to make type casting so that the transpiler doesn't throw errors. It's just a matter of where this transformation/casting should be done, service or component. The approach is to provide unified errors to display, which in this case, needs the same fields hence type casting for errors